core/utils/style.ts

```diff
@@ -292,3 +269,34 @@ export function getContainerOffsetToScrollInto(
   }
   return new Coordinate(scrollLeft, scrollTop);
 }
+
+function getSizeInternal(element: Element): Size {
+  if (getStyle(element, 'display') !== 'none') {
+    return getSizeWithDisplay(element);
+  }
+
+  // Evaluate size with a temporary element.
+  // AnyDuringMigration because:  Property 'style' does not exist on type
+  // 'Element'.
+  const style = (element as AnyDuringMigration).style;
+  const originalDisplay = style.display;
+  const originalVisibility = style.visibility;
+  const originalPosition = style.position;
+
+  style.visibility = 'hidden';
+  style.position = 'absolute';
+  style.display = 'inline';
+
+  const offsetWidth = (element as HTMLElement).offsetWidth;
+  const offsetHeight = (element as HTMLElement).offsetHeight;
+
+  style.display = originalDisplay;
+  style.position = originalPosition;
+  style.visibility = originalVisibility;
+
+  return new Size(offsetWidth, offsetHeight);
+}
+
+export const TEST_ONLY = {
+  getSizeInternal,
+}
```

Comment:
Missing `\n` at EOF.  Does your editor have an option analogous to emacs's `require-final-newline`?